resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/Client.scala

```diff
@@ -1485,6 +1486,22 @@ private object Client extends Logging {
     YarnAppReport(report.getYarnApplicationState(), report.getFinalApplicationStatus(), diagsOpt)
   }
 
+  /**
+   * Create a properly quoted library path string to be added as a prefix to the command executed by
+   * YARN. This is different from plain quoting due to YARN executing the command through "bash -c".
+   */
+  def createLibraryPathPrefix(libpath: String, conf: SparkConf): String = {
```

Comment:
This is so specific to the way YARN runs things that I don't think it would be useful anywhere else. If at some point it becomes useful, the code can be moved.

I think the tests I added are better than just unit testing this function, since that way the code is actually being run through YARN and bash.